Conventional/Conventions/PropertiesMustHavePublicGettersConventionSpecification.cs

```diff
@@ -1,28 +1,18 @@
 using System;
 using System.Linq;
+using System.Reflection;
 
 namespace Conventional.Conventions
 {
-    public class PropertiesMustHavePublicGettersConventionSpecification : ConventionSpecification
+    public class PropertiesMustHavePublicGettersConventionSpecification : PropertyConventionSpecification
     {
         protected override string FailureMessage => "All properties must have public getters";
 
-        public override ConventionResult IsSatisfiedBy(Type type)
+        protected override PropertyInfo[] GetNonConformingProperties(Type type)
         {
-            var toInspect = type.GetProperties().Where(p => p.CanWrite);
-
-            var failures = toInspect.Where(subject => subject.GetGetMethod() == null || subject.GetGetMethod().IsPublic == false).ToArray();
-
-            if (failures.Any())
-            {
-                var failureMessage =
-                    BuildFailureMessage(failures.Aggregate(string.Empty,
-                        (s, info) => s + "\t- " + info.Name + Environment.NewLine));
-
-                return ConventionResult.NotSatisfied(type.FullName, failureMessage);
-            }
-
-            return ConventionResult.Satisfied(type.FullName);
+            return type.GetProperties()
+                .Where(subject => subject.CanRead == false || subject.GetGetMethod(true).IsPublic == false)

```

Comment:
Woah, I had no idea that a prop with a private getter returns `true` for CanRead! Barmy! Guessing that's why this checks both conditions?